ui/views.py

```diff
@@ -154,6 +157,32 @@ def get_context_data(self, video, **kwargs):  # pylint: disable=arguments-differ
         return context
 
 
+class VideoDownload(VideoDetail):
+    """
+    Download a public video
+    """
+    def download(self, video):
+        """
+        Return the response as a file download
+        Returns:
+            HttpResponse: the response videofile content and a file attachment header
+        """
+        video_file = video.download
+        if not video_file:
+            raise Http404()
+        _, ext = splitext(video_file.s3_object_key)
+        video_bytes = requests.get(video_file.cloudfront_url, stream=True)
+        response = StreamingHttpResponse(
+            (chunk for chunk in video_bytes.iter_content(512 * 1024)),
```

Comment:
Would just passing the `iter_content(...)` call directly work here, instead of wrapping in a generator?

Reply:
Yes, that seems to work